server/src/test/java/io/crate/lucene/CommonQueryBuilderTest.java

```diff
@@ -749,4 +749,11 @@ public void test_neq_operator_on_nullable_and_not_nullable_args_filters_nulls()
             assertThat(tester.runQuery("a", "a != a||1")).containsExactly(1, 2);
         }
     }
+
+    // tracks a bug: https://github.com/crate/crate/issues/15232
+    @Test
+    public void test_cannot_use_field_exists_query_on_args_of_coalesce_function() {
+        Query query = convert("coalesce(x, y) <> 0");
+        assertThat(query).hasToString("+(+*:* -(coalesce(x, y) = 0))");
```

Comment:
The query before fix : `+(+*:* -(coalesce(x, y) = 0)) +FieldExistsQuery [field=y]` (where FieldExistsQuery for `x` was missing because it was `not null`)